tests/functional/horizon_controller_test.go

```diff
@@ -145,4 +146,21 @@ var _ = Describe("Horizon controller", func() {
 			Expect(memcached).NotTo(BeNil())
 		})
 	})
+	When("Route Annotation is supplied", func() {
+		BeforeEach(func() {
+			DeferCleanup(DeleteInstance, CreateSharedMemcached())
+			DeferCleanup(DeleteInstance, CreateHorizonSharedMemcachedWithAnnotation(horizonName))
+			secret = &corev1.Secret{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      SecretName,
+					Namespace: namespace,
+				},
+			}
+			Expect(k8sClient.Create(ctx, secret)).Should(Succeed())
+		})
+		It("Should have an Annotation on the Route", func() {
+			horizonRoute := k8sClient.List(ctx, &routev1.RouteList{})
+			fmt.Printf("debug: HorizonRoute: %+v", horizonRoute)
+		})
+	})
```

Comment:
I don't think this will be possible with Envtest. The Route API is provided via the `openshift-api` server using the API Aggregation Kubernetes method:
https://kubernetes.io/docs/concepts/extend-kubernetes/api-extension/apiserver-aggregation/

So, unless we start Envtest and install the Openshift-api server, I don't know that we will have access to any of the custom resources provided by Openshift-api.

Reply:
Again, this is where it would be easier to just use Ingress...